Refactor Docs

k4lizen opened the pull request:
This PR will track refactoring docs generation/build/mkdocs.yml settings.

Current changes:
+ The generated Source page looks better
![image](https://github.com/user-attachments/assets/575db008-a850-45b6-904c-566a508edb9d)

+ The commands page looks better
![image](https://github.com/user-attachments/assets/5bf50156-4ad6-4e2f-9e2a-f3f49a8e6485)


+ The navigation is autogenerated so we don't have to remember to add a line to `mkdocs.yml` when writing docs
+ The folders are restructured so they make more sense for the docs page rather than mirror the source layout

k4lizen committed:
add comments in mkdocs, autogenerate nav

k4lizen committed:
reorganize the docs folder structure so they look good when the nav is autogenerated

k4lizen committed:
add really cool Source page

k4lizen committed:
fix edit, hide nav in setup, cleanup mkdocstrings config a bit

k4lizen committed:
fix doc CI build failure

patryk4815 commented on mkdocs.yml:
maybe because of that `--only-group` don't work

k4lizen replied:
I can confirm that disabling api-autonav allows `--only-group` to work, but I'm not sure why. It happens even when I set `allow_inspection: false` which should prevent any module loading, so to my understanding it should just be statically analyzing the files.

disconnect3d commented:
(Fwiw if that makes sense, we can also merge the docs refactorings in parts)

k4lizen committed:
cleanup mkdocs.yml

k4lizen committed:
allow mkdocstrings to use linter

k4lizen committed:
remove empty files

k4lizen commented:
I think this is mostly it relating to what the docs look like (I'll handle other stuff in other PRs), so it should be ready for review.

Can test the PR with (it takes a bit to spin up):
```
uv run --group docs mkdocs serve
```

Feedback welcome obviously :p

k4lizen committed:
show parameters cleaner

according to the google style guide, we shouldn't specify types when they are in annotation, so using table which always shows types doesn't make sense

disconnect3d commented:
Merging this. Let's see this on prod and we will give feedback based on the result on Discord, then we can have more PRs improving this further.
